source/opt/ir_context.h

```diff
@@ -218,16 +220,28 @@ class IRContext {
   // Invalidates the analyses marked in |analyses_to_invalidate|.
   void InvalidateAnalyses(Analysis analyses_to_invalidate);
 
-  // Turns the instruction defining the given |id| into a Nop. Returns true on
+  // Deletes the instruction defining the given |id|. Returns true on
   // success, false if the given |id| is not defined at all. This method also
-  // erases both the uses of |id| and the information of this |id|-generating
-  // instruction's uses of its operands.
+  // erases the name, decorations, and defintion of |id|.
+  //
+  // Pointers and iterators pointing to the deleted instructions become invalid.
+  // However other pointers and iterators are still valid.
   bool KillDef(uint32_t id);
```

Comment:
Would it make sense to make KillDef return the same value as KillInstr?  I'm not sure the suggestion makes sense.  Just thought it may be useful.

Reply:
It already has its own return value.  I did not check, but I'm guessing it used used for a reason.